File: Easy/0020 - validparentheses/validparentheses.cpp
# include <string>
# include <stack>
using namespace std;

class Solution {
    public:
        bool isValid(string s)
        {
            stack<char> stack;

            for (int i = 0; i < s.length(); i++)
            {
                if (s[i] == '(' || s[i] == '{' || s[i] == '[')
                    stack.push(s[i]);
                else {
                    if (stack.empty())
                        return false;

                    char top = stack.top();
                    if ((s[i] == ')' && top == '(') ||
                        (s[i] == '}' && top == '{') ||
                        (s[i] == ']' && top == '['))
                            stack.pop();
                    else
                        return false;
                }
            }

            return stack.empty();
        }
};

File: Easy/0020 - validparentheses/validparentheses.js
var isValid = function(s) {
    let stack = [];
    
    for (let char of s)
    {
        if (char === '(' || char === '{' || char === '[')
            stack.push(char);
        else
        {
            if (stack.length === 0)
                return false;
            
            let top = stack[stack.length - 1];
            if ((char === ')' && top === '(') ||
                (char === '}' && top === '{') ||
                (char === ']' && top === '['))
                    stack.pop();
            else
                return false;
        }
    }
    
    return stack.length === 0;
};